README.md

```diff
@@ -1,63 +1,3280 @@
+```
+Finally, push the code in the `multicontainer-app` directory to the application:
+```
+$ cd $BALENA_DEBUGGING_MASTERCLASS/multicontainer-app
+$ balena push DebugApp
+```
 
-This is a subtopic exercise section for a topic.
+## 1. Accessing a User Device
 
-## 1.2 Subtopic 1.2 Heading
+Any device owned by a customer automatically allows access by that user via
+either the WebTerminal (in the device's Dashboard page), or via the balena CLI
+via `balena ssh <uuid>`. However, for a support agent to gain access to a device
+that isn't owned by them, a user or collaborator that does have access must
+grant it explicitly.
 
-This is another subtopic exercise section for a topic.
+### 1.1 Granting Support Access to a Support Agent
 
-# 2. Topic 2 Heading
+A user can grant access for a device to a support agent by selecting the device
+they wish to grant access to from the Dashboard, and then selecting the
+'Actions' tab in the left-hand sidepanel. Scrolling down the Actions page will
+show a list of actions, with the 'Grant Support Access' option being the one
+required here. A user can select this, then detemine the amount of time that
+support agents are allowed access for. Once support has been granted, the
+Dashboard will look something like this:
+![Granted Support Access](resources/black-mountain-granted.png)
 
-This is the second topic exercise section.
+**Note:** It's also possible for a user to grant support for an entire
+application by selecting the application's Dashboard page and going through
+the same process (selecting 'Actions' and then 'Grant Support Access'). Granting
+access to an application automatically grants access to all of its associated
+devices.
 
-# Conclusion
+Once support access has been granted, an agent will be able to use the UUID of
+a device to gain access to it, using a URL of the form:
+https://dashboard.balena-cloud.com/devices/<deviceUUID>/summary
+
+The Dashboard will function in almost exactly in the same way as it would were
+the device owned by the support agent viewing it. They may view logs and
+use the WebTerminal to access either the Host balenaOS or any service currently
+running.
+
+They may also use balena CLI to SSH into either the balenaOS host or any service
+using `balena ssh <uuid> [serviceName]`.
+
+### 1.2 Access Restrictions
+
+There are limits on what a support agent may do with a device they have
+been granted access to. This includes the alteration of service and environment
+variables and configurations (both application and device).
+
+Whilst this sounds like a limitation, it in fact ensures that a device being
+investigated for an issue cannot be unduly altered or modified. Support
+investigations are intended as an avenue of exploration and research for
+ensuring that issues are categorised to allow improvements to the product
+surface such that these issues are eliminated.
+
+## 2. Initial Diagnosis
+
+The balenaCloud Dashboard includes the ability to run a set of diagnostics on
```

Comment:
maybe there's bloggable content here along with https://github.com/balena-io/device-diagnostics/issues/163?

Reply:
Yeah, I definitely think so. Maybe you (and I if you want a hand?) should write something. cc @shaunmulligan @chrisys @garethtdavies